public/javascripts/script.js

```diff
@@ -1,4889 +0,0 @@
-  }
-}
-
-// Below, we have frontend form validation that prevents a user from trying to sign up with an invalid password, enhancing security. 
-
-function passwordPass(e) {
-  const password = document.getElementById("password").value;
-  const regex = /(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{6,}/;
-  if (password.length > 4) {
-    if (!regex.test(password)) {
-      document.getElementById("passwordError").classList.remove("invisible", "h-0");
-      document.getElementById("passwordError").classList.add("pt-2");
-    } else {
-      document.getElementById("passwordError").classList.add("invisible", "h-0");
-      document.getElementById("passwordError").classList.remove("pt-2");
-    }
-  }
-}
-
-// Below, I'm handling all of the filters on the homepage that allows users to filter all of the companies that have been reviewed by their different aspects. There's 5 different event listeners for the 5 filters. Whenever one is selected, it toggles the checkedFilter booleans and runs a function that takes in those checked filters. This allows users to combine different filters and see only those that meet all of their requirements. Neat!
-
-const checkedFilters = {
-  easy: false,
-  features: false,
-  customerSupport: false,
-  valueForMoney: false,
-  distribution: false
-};
-
-document.getElementById("featureEasy").addEventListener("click", () => {
-  if (!checkedFilters.easy) {
-    checkedFilters.easy = true;
-  } else {
-    checkedFilters.easy = false;
-  }
-  document.getElementById("featureEasy").classList.toggle("bg-blue-100");
-  document.getElementById("featureEasy").classList.toggle("border-blue-600");
-  document.getElementById("featureEasy").classList.toggle("border-solid");
-  document.getElementById("featureEasy").classList.toggle("border-4");
-  runFilters(checkedFilters);
-});
-
-document.getElementById("featureFeature").addEventListener("click", () => {
-  if (!checkedFilters.features) {
-    checkedFilters.features = true;
-  } else {
-    checkedFilters.features = false;
-  }
-  document.getElementById("featureFeature").classList.toggle("bg-blue-100");
-  document.getElementById("featureFeature").classList.toggle("border-blue-600");
-  document.getElementById("featureFeature").classList.toggle("border-solid");
-  document.getElementById("featureFeature").classList.toggle("border-4");
-  runFilters(checkedFilters);
-});
-
-document.getElementById("featureSupport").addEventListener("click", () => {
-  if (!checkedFilters.customerSupport) {
-    checkedFilters.customerSupport = true;
-  } else {
-    checkedFilters.customerSupport = false;
-  }
-  document.getElementById("featureSupport").classList.toggle("bg-blue-100");
-  document.getElementById("featureSupport").classList.toggle("border-blue-600");
-  document.getElementById("featureSupport").classList.toggle("border-solid");
-  document.getElementById("featureSupport").classList.toggle("border-4");
-  runFilters(checkedFilters);
```

Comment:
Again, this is copy-pasted from above function - let's abstract this out to a function